kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/TableChangesSuite.scala

```diff
@@ -0,0 +1,400 @@
+      }.getMessage.contains("no log file found for version 8"))
+
+      // invalid start version
+      assert(intercept[KernelException]{
+        getChangesByVersion(-1, 2)
+      }.getMessage.contains("Invalid version range"))
+
+      // invalid end version
+      assert(intercept[KernelException]{
+        getChangesByVersion(2, 1)
+      }.getMessage.contains("Invalid version range"))
+    }
+  }
+
+  test("getChangesByVersion - with truncated log") {
+    withTempDir { tempDir =>
+      // PREPARE TEST TABLE
+      val tablePath = tempDir.getCanonicalPath
+      // Write versions [0, 10] (inclusive) including a checkpoint
+      (0 to 10).foreach { i =>
+        spark.range(i*10, i*10 + 10).write
+          .format("delta")
+          .mode("append")
+          .save(tablePath)
+      }
+      val log = org.apache.spark.sql.delta.DeltaLog.forTable(
+        spark, new org.apache.hadoop.fs.Path(tablePath))
+      val deltaCommitFileProvider = org.apache.spark.sql.delta.util.DeltaCommitFileProvider(
+        log.unsafeVolatileSnapshot)
+      // Delete the log files for versions 0-9, truncating the table history to version 10
+      (0 to 9).foreach { i =>
+        val jsonFile = deltaCommitFileProvider.deltaFile(i)
+        new File(new org.apache.hadoop.fs.Path(log.logPath, jsonFile).toUri).delete()
+      }
+      // Create version 11 that overwrites the whole table
+      spark.range(50).write
+        .format("delta")
+        .mode("overwrite")
+        .save(tablePath)
+      // Create version 12 that appends new data
+      spark.range(10).write
+        .format("delta")
+        .mode("append")
+        .save(tablePath)
+
+      // TEST ERRORS
+      // endVersion before earliest available version
+      assert(intercept[KernelException] {
+        Table.forPath(defaultEngine, tablePath)
+          .asInstanceOf[TableImpl]
+          .getChangesByVersion(defaultEngine, 0, 9, FULL_ACTION_SET.asJava)
+      }.getMessage.contains("no log files found in the requested version range"))
+
+      // startVersion less than the earliest available version
+      assert(intercept[KernelException] {
+        Table.forPath(defaultEngine, tablePath)
+          .asInstanceOf[TableImpl]
+          .getChangesByVersion(defaultEngine, 5, 11, FULL_ACTION_SET.asJava)
+      }.getMessage.contains("no log file found for version 5"))
+
+      // TEST VALID CASES
+      testGetChangesByVersionVsSpark(
+        tablePath,
+        10,
+        12,
+        FULL_ACTION_SET
+      )
+      testGetChangesByVersionVsSpark(
+        tablePath,
+        11,
+        12,
+        FULL_ACTION_SET
+      )
+    }
+  }
+
+  //////////////////////////////////////////////////////////////////////////////////
+  // Helpers to compare actions returned between Kernel and Spark
+  //////////////////////////////////////////////////////////////////////////////////
+
+  // Standardize actions with case classes, keeping just a few fields to compare
```

Comment:
For class docs, use `/** */` btw.

Also, please give. some more context here. What does "standardize" really mean? i.e. we are doing this so that we can map kernel actions and spark actions to the same data types (StandardAdd and StandardRemove) so that we can compare them easily